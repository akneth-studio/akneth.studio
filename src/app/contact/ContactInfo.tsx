'use client'

import React from 'react';
import Image from 'next/image';
import { LiaMapMarkedSolid, LiaPhoneSolid, LiaAtSolid } from 'react-icons/lia';

// Import environment variables for contact information
const street = process.env.NEXT_PUBLIC_ADDRESS_STREET || 'ul. PrzykÅ‚adowa 1';
const city = process.env.NEXT_PUBLIC_ADDRESS_CITY || 'Warszawa';
const phone = process.env.NEXT_PUBLIC_PHONE || '123-456-789';
const email = process.env.NEXT_PUBLIC_EMAIL || 'kontakt@przyklad.pl';

import { Autour_One } from 'next/font/google';
const autourOne = Autour_One({
    weight: '400',
    subsets: ['latin-ext'],
    display: 'swap',
});

const ContactInfo = () => {
    return (
        <>
            <Image
                src="/img/logo_akneth.svg"
                alt="AKNETH Studio"
                width={300}
                height={150}
            />
            <div className="mb-4">
                <h4><LiaMapMarkedSolid /> Adres</h4>
                <div className="mb-2">
                    <p>
                        {street} <br />
                        {city}
                    </p>
                </div>
                <h4><LiaPhoneSolid /> Telefon</h4>
                <p>{phone}</p>
                <h4><LiaAtSolid /> Email</h4>
                <p>{email}</p>
            </div>
        </>
    );
};

export default ContactInfo;
