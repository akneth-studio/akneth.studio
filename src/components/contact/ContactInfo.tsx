'use client'

import React from 'react';
import Image from 'next/image';
import { LiaMapMarkedSolid, LiaPhoneSolid, LiaAtSolid, LiaClockSolid } from 'react-icons/lia';

// Import environment variables for contact information
const street = process.env.NEXT_PUBLIC_ADDRESS_STREET || 'ul. Przykładowa 1';
const city = process.env.NEXT_PUBLIC_ADDRESS_CITY || 'Warszawa';
const phone = process.env.NEXT_PUBLIC_PHONE || '123-456-789';
const email = process.env.NEXT_PUBLIC_EMAIL || 'kontakt@przyklad.pl';

const ContactInfo = () => {
    return (
        <>
            <Image
                src="/img/logo_akneth.svg"
                alt="AKNETH Studio"
                width={300}
                height={150}
            />
            <div className="mb-4">
                <h4><LiaMapMarkedSolid /> Adres</h4>
                <div className="mb-2">
                    <p>
                        {street} <br />
                        {city}
                    </p>
                </div>
                <h4><LiaClockSolid /> Godziny pracy</h4>
                <p>
                    <b>Poniedziałek - Piątek</b><br/>
                    9:00 - 17:00
                </p>
                <p>
                    <b>Sobota</b><br/>
                    10:00 - 14:00<br/>
                    <span className='fst-italic text-body-secondary' style={{fontSize: '0.85rem'}}>
                        (tylko umówione spotkania)
                    </span>
                </p>
                <h4><LiaPhoneSolid /> Telefon</h4>
                <p>{phone}</p>
                <h4><LiaAtSolid /> Email</h4>
                <p>{email}</p>
            </div>
        </>
    );
};

export default ContactInfo;
