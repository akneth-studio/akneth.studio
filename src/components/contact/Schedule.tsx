'use client';

import React from 'react';

const scheduleUrl = process.env.NEXT_PUBLIC_SCHEDULE_URL;

export default function Schedule() {
    if (!scheduleUrl) {
        return (
            <div className="alert alert-warning text-center my-3" style={{ fontSize: '0.95rem', color: '#738a96' }} role="alert">
                <p>Harmonogram spotkań jest chwilowo niedostępny. Skontaktuj się mailowo <a href="mailto:akneth.studio@gmail.com">akneth.studio@gmail.com</a> lub skorzystaj z formularza kontaktowego.</p>
            </div>
        );
    }

    return (
        <section aria-label="Harmonogram spotkań" style={{ width: '100%' }} className="schedule my-5">
            <div style={{ position: 'relative', paddingBottom: '70%', height: 0, overflow: 'hidden', borderRadius: '1rem', boxShadow: '0 2px 16px rgba(52,58,64,0.08)' }}>
                <iframe
                    src={scheduleUrl}
                    title="Umów spotkanie"
                    width="100%"
                    height="100%"
                    style={{
                        position: 'absolute',
                        top: 0,
                        left: 0,
                        width: '100%',
                        height: '100%',
                        border: 'none',
                        borderRadius: '1rem',
                        background: '#fff',
                    }}
                    loading="lazy"
                    allow="clipboard-write"
                />
            </div>
        </section>
    );
}
