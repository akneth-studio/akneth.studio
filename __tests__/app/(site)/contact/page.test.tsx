/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import ContactPage from '@/app/(site)/contact/page';
import ContactFormWrapper from '@/components/contact/ContactForm';
import Schedule from '@/components/contact/Schedule';
import ContactInfo from '@/components/contact/ContactInfo';

// Mock the components
jest.mock('@/components/contact/ContactForm', () => {
  return jest.fn(() => <div data-testid="contact-form">Contact Form</div>);
});

jest.mock('@/components/contact/Schedule', () => {
  return jest.fn(() => <div data-testid="schedule">Schedule</div>);
});

jest.mock('@/components/contact/ContactInfo', () => {
  return jest.fn(() => <div data-testid="contact-info">Contact Info</div>);
});

// Mock environment variables
process.env.NEXT_PUBLIC_SITE_URL = require('../../../../jest.setup').NEXT_PUBLIC_SITE_URL || 'https://akneth.pl';
process.env.NEXT_PUBLIC_RECAPTCHA_SITE_KEY = 'test-recaptcha-key';

describe('ContactPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render all sections correctly when reCAPTCHA key is present', () => {
    render(<ContactPage />);
    
    // Check if all sections are rendered
    expect(screen.getByTestId('contact-info')).toBeInTheDocument();
    expect(screen.getByTestId('contact-form')).toBeInTheDocument();
    expect(screen.getByTestId('schedule')).toBeInTheDocument();
    
    // Check if the heading exists
    expect(screen.getByText('Kontakt')).toBeInTheDocument();
  });

  it('should render error message when reCAPTCHA key is missing', () => {
    // Temporarily remove the reCAPTCHA key
    const originalKey = process.env.NEXT_PUBLIC_RECAPTCHA_SITE_KEY;
    delete process.env.NEXT_PUBLIC_RECAPTCHA_SITE_KEY;
    
    render(<ContactPage />);
    
    // Check if error message is displayed
    expect(screen.getByText('Błąd konfiguracji reCAPTCHA. Skontaktuj się z administratorem strony.')).toBeInTheDocument();
    expect(screen.getByTestId('schedule')).toBeInTheDocument();
    
    // Restore the reCAPTCHA key
    process.env.NEXT_PUBLIC_RECAPTCHA_SITE_KEY = originalKey;
  });

  it('should render the correct layout structure', () => {
    render(<ContactPage />);
    
    // Check if the main container exists (using the heading as reference)
    const heading = screen.getByText('Kontakt');
    expect(heading).toBeInTheDocument();
    expect(heading.tagName).toBe('H1');
    
    // Check if the row structure exists
    const row = screen.getByTestId('contact-info').parentElement?.parentElement;
    expect(row).toBeInTheDocument();
    expect(row).toHaveClass('row');
    
    // Check if the columns exist
    const contactInfoColumn = screen.getByTestId('contact-info').parentElement;
    expect(contactInfoColumn).toBeInTheDocument();
    expect(contactInfoColumn).toHaveClass('col-md-6');
    
    const contactFormColumn = screen.getByTestId('contact-form').parentElement;
    expect(contactFormColumn).toBeInTheDocument();
    expect(contactFormColumn).toHaveClass('col-md-6');
  });

  it('should have correct metadata', () => {
    // Check if metadata is defined
    const { metadata } = require('@/app/(site)/contact/page');
    expect(metadata).toBeDefined();
    expect(metadata.title).toBe('Kontakt');
    expect(metadata.description).toBe('Skontaktuj się z AKNETH Studio');
    expect(metadata.keywords).toEqual([
      'AKNETH Studio',
      'Kontakt',
      'Harmonogram',
      'Formularz Kontaktowy'
    ]);
    expect(metadata.alternates?.canonical).toBe('https://akneth.pl/contact');
  });

  it('should match snapshot', () => {
    const { container } = render(<ContactPage />);
    expect(container).toMatchSnapshot();
  });
});
