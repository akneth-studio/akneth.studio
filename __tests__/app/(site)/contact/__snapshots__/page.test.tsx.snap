// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ContactPage should match snapshot 1`] = `
<div>
  <h1
    class="mb-4 pt-5 text-center"
  >
    Kontakt
  </h1>
  <div
    class="row"
  >
    <section
      class="col-md-6 mb-4 mb-md-0 contact-info text-center align-md-start align-content-center"
    >
      <div
        data-testid="contact-info"
      >
        Contact Info
      </div>
    </section>
    <aside
      class="col-md-6"
    >
      <div
        data-testid="contact-form"
      >
        Contact Form
      </div>
    </aside>
  </div>
  <div
    data-testid="schedule"
  >
    Schedule
  </div>
</div>
`;
